src/internal/connector/graph/errors.go

```diff
@@ -172,6 +173,10 @@ func IsMalwareResp(ctx context.Context, resp *http.Response) bool {
 	return false
 }
 
+func IsErrFolderExists(err error) bool {
```

Comment:
Please add some unit tests for this.  There should be some easy boilerplate to steal from.